packages/jest-cli/src/SearchSource.js

```diff
@@ -183,6 +185,20 @@ class SearchSource {
     };
   }
 
+  findRelatedTestsFromPattern(
+    patternInfo: PatternInfo,
+    cwd: Path = process.cwd(),
+  ): SearchResult {
+    if (patternInfo.input.length) {
+      const sources = patternInfo.input.split(' ');
```

Comment:
I'm afraid this will break for inputs like `jest --findRelatedTests test.js "file with spaces.js"` because it would join them first and then split them again.

I think instead we should treat `input` as an array here: https://github.com/facebook/jest/blob/deacf566ff079a441f46157a4af55c899047d401/packages/jest-cli/src/jest.js#L111 which might require some other changes in SearchResult.js or jest.js